deeplearning4j-nn/src/main/java/org/deeplearning4j/datasets/iterator/AsyncDataSetIterator.java

```diff
@@ -113,291 +91,282 @@ public AsyncDataSetIterator(DataSetIterator iterator, int queueSize, BlockingQue
 
+    /**
+     * Input columns for the dataset
+     *
+     * @return
+     */
     @Override
     public int inputColumns() {
-        return baseIterator.inputColumns();
+        return backedIterator.inputColumns();
     }
 
+    /**
+     * The number of labels for the dataset
+     *
+     * @return
+     */
     @Override
     public int totalOutcomes() {
-        return baseIterator.totalOutcomes();
+        return backedIterator.totalOutcomes();
     }
 
+    /**
+     * Is resetting supported by this DataSetIterator? Many DataSetIterators do support resetting,
+     * but some don't
+     *
+     * @return true if reset method is supported; false otherwise
+     */
     @Override
     public boolean resetSupported() {
-        return baseIterator.resetSupported();
+        return backedIterator.resetSupported();
     }
 
+    /**
+     * Does this DataSetIterator support asynchronous prefetching of multiple DataSet objects?
+     * Most DataSetIterators do, but in some cases it may not make sense to wrap this iterator in an
+     * iterator that does asynchronous prefetching. For example, it would not make sense to use asynchronous
+     * prefetching for the following types of iterators:
+     * (a) Iterators that store their full contents in memory already
+     * (b) Iterators that re-use features/labels arrays (as future next() calls will overwrite past contents)
+     * (c) Iterators that already implement some level of asynchronous prefetching
+     * (d) Iterators that may return different data depending on when the next() method is called
+     *
+     * @return true if asynchronous prefetching from this iterator is OK; false if asynchronous prefetching should not
+     * be used with this iterator
+     */
     @Override
     public boolean asyncSupported() {
         return false;
     }
 
+    /**
+     * Resets the iterator back to the beginning
+     */
     @Override
-    public synchronized void reset() {
-        if (!resetSupported())
-            throw new UnsupportedOperationException(
-                    "Cannot reset Async iterator wrapping iterator that does not support reset");
-        //Complication here: runnable could be blocking on either baseIterator.next() or blockingQueue.put()
-        runnable.killRunnable = true;
-        if (runnable.isAlive.get()) {
-            thread.interrupt();
-        }
-        //Wait for runnable to exit, but should only have to wait very short period of time
-        //This probably isn't necessary, but is included as a safeguard against race conditions
-        try {
-            runnable.runCompletedSemaphore.tryAcquire(5, TimeUnit.SECONDS);
-        } catch (InterruptedException e) {
-        }
-
-        //Clear the queue, reset the base iterator, set up a new thread
-        blockingQueue.clear();
-        baseIterator.reset();
-        runnable = new IteratorRunnable(baseIterator.hasNext(), this.workspace);
-        thread = runnable;
+    public void reset() {
```

Comment:
No, it doesn't. ADSI isn't (and wasn't) supposed to be thread-safe, training is single-threaded by design.